src/bootstrap/install.rs

```diff
@@ -139,11 +139,15 @@ macro_rules! install {
 
 install!((self, builder, _config),
     Docs, "src/doc", _config.docs, only_hosts: false, {
+        // `expect` should be safe, only None when config.docs is false,
+        // which is guarded in `should_run`
```

Comment:
I think x.py install src/doc will still fail here -- the guard in the dist step is part of run(), not whether we run it by default, so it will return None even if we're not running this step by default.

Reply:
I tested `./x.py install` locally with docs=false, and it seemed to be fine.  The guard I'm referring to is [here](https://github.com/rust-lang/rust/blob/3a24abd22fd25c836d8b4d75ff46c833f9c3934c/src/bootstrap/install.rs#L121-L124) where it expands to `run.path("src/doc").default_condition(_config.docs)`.  The `default_condition` should disable it when docs=false, correct?

Reply:
Ah, I see.  I've added a better error message for that situation.